backends/beats/configuration.go

```diff
@@ -141,3 +155,64 @@ func (bc *BeatsConfig) PropertyBool(p interface{}) bool {
 		return false
 	}
 }
+
+func (bc *BeatsConfig) RunMigrations() {
+	if bc.Version[0] == 5 && bc.Version[1] == 0 {
+		// rename ssl properties
+		cp := bc.Container
+		configurationPath := []string{"output", "logstash", "tls", "certificate_key"}
+		for target := 0; target < len(configurationPath); target++ {
+			if mmap, ok := cp.(map[string]interface{}); ok {
+				if target == len(configurationPath)-1 {
+					if mmap["certificate_key"] != nil {
+						mmap["key"] = mmap["certificate_key"]
+						delete(mmap, "certificate_key")
+					}
+					if mmap["insecure"] == true {
+						mmap["verification_mode"] = "none"
+						delete(mmap, "insecure")
+					}
+				}
+				cp = mmap[configurationPath[target]]
+			}
+		}
+
+		// rename tls -> ssl
+		cp = bc.Container
+		configurationPath = []string{"output", "logstash", "tls"}
+		for target := 0; target < len(configurationPath); target++ {
+			if mmap, ok := cp.(map[string]interface{}); ok {
+				if target == len(configurationPath)-1 {
+					if mmap["tls"] != nil {
+						mmap["ssl"] = mmap["tls"]
+						delete(mmap, "tls")
+					}
+				}
+				cp = mmap[configurationPath[target]]
+			}
+		}
+
+		// remove shipper
+		cp = bc.Container
+		configurationPath = []string{"shipper", "tags"}
+		for target := 0; target < len(configurationPath); target++ {
+			if mmap, ok := cp.(map[string]interface{}); ok {
+				if target == len(configurationPath)-1 {
+					bc.Set(mmap["tags"], "tags")
+					delete(bc.Container.(map[string]interface{}), "shipper")
+				}
+				cp = mmap[configurationPath[target]]
+			}
+		}
+
+		// set cache data path
+		dataPath := bc.UserConfig.RunPath
+		if dataPath == "" {
+			dataPath = filepath.Join(filepath.Dir(bc.UserConfig.BinaryPath), "data")
```

Comment:
This will break once someone sets the path to the collector to `/usr/bin/filebeat`. Then it tries to create `/usr/bin/data`.

Isn't there a central state directory for the sidecar where the collectors can write to?